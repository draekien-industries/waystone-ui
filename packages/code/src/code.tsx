import type { PropsWithChildren } from 'react';
import { codeCss } from './code.styles';

export const Code = ({ children }: PropsWithChildren) => (
  <code sx={codeCss}>{children}</code>
);
