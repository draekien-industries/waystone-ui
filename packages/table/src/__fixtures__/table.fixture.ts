import { faker } from '@faker-js/faker';
import type { ColumnDef } from '@tanstack/react-table';
import { createInteractiveColumn } from '../components/prefabs';

type Person = {
  firstName: string;
  lastName: string;
  age: number;
  visits: number;
  progress: number;
  status: 'relationship' | 'complicated' | 'single';
  subRows?: Person[];
};

const range = (len: number) => {
  const arr: number[] = [];
  for (let i = 0; i < len; i++) {
    arr.push(i);
  }
  return arr;
};

faker.seed(459201);

const newPerson = (): Person => ({
  firstName: faker.person.firstName(),
  lastName: faker.person.lastName(),
  age: faker.number.int(40),
  visits: faker.number.int(1000),
  progress: faker.number.int(100),
  status: faker.helpers.shuffle<Person['status']>([
    'relationship',
    'complicated',
    'single',
  ])[0]!,
});

function makeData(...lens: number[]) {
  const makeDataLevel = (depth = 0): Person[] => {
    const len = lens[depth]!;
    return range(len).map(
      (): Person => ({
        ...newPerson(),
        subRows: lens[depth + 1] ? makeDataLevel(depth + 1) : undefined,
      })
    );
  };

  return makeDataLevel();
}

export const columns: ColumnDef<Person>[] = [
  createInteractiveColumn<Person>(),
  {
    accessorKey: 'firstName',
    header: 'First Name',
  },
  {
    accessorKey: 'lastName',
    header: 'Last Name',
  },
  {
    accessorKey: 'age',
    header: 'Age',
  },
  {
    accessorKey: 'visits',
    header: 'Visits',
  },
  {
    accessorKey: 'status',
    header: 'Status',
  },
  {
    accessorKey: 'progress',
    header: 'Progress',
  },
];

export const data = makeData(20, 5, 3);
