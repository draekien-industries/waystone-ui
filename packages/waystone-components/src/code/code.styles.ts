import type { ThemeUIStyleObject } from 'theme-ui';

export const codeCss = (darkMode: boolean): ThemeUIStyleObject => ({
  variant: 'text.mono',
  fontSize: 'caption',
  backgroundColor: darkMode ? 'b-600' : 'b-200',
  borderRadius: 'sm',
  px: 'xs',
  py: 2,
  color: darkMode ? 'highlight' : 'b-600',
});
