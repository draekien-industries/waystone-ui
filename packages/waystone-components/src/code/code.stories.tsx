import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { Code } from './code';
import { Text } from '../text/text';

export default {
  title: 'Components/Code',
  component: Code,
} as ComponentMeta<typeof Code>;

const Template: ComponentStory<typeof Code> = (args) => <Code {...args} />;

export const Default = Template.bind({});

Default.args = {
  children: 'code',
};

export function WithExample() {
  return (
    <Text>
      This is an example of the <Code>Code</Code> component being used inside a{' '}
      <Code>Text</Code> component.
    </Text>
  );
}
