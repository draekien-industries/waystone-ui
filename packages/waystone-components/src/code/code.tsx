'use client';

import { ReactNode } from 'react';
import { useColorMode } from 'theme-ui';
import { codeCss } from './code.styles';

/** The props required to render the code component. */
export interface CodeProps {
  /** The code to render. */
  children: ReactNode;
}

/**
 * An inline code snippet.
 */
export const Code = ({ children }: CodeProps) => {
  const [mode] = useColorMode();

  return <code sx={codeCss(mode === 'dark')}>{children}</code>;
};
