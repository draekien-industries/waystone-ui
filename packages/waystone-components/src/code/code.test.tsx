import React from 'react';
import { render } from '../__fixtures__';
import { Code } from './code';

describe('The Code component', () => {
  it('Matches snapshot', () => {
    const { container } = render(<Code>code</Code>);

    expect(container).toMatchSnapshot();
  });
});
