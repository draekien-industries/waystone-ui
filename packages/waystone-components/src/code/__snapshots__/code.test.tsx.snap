// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Code component Matches snapshot 1`] = `
<div>
  <code
    class="css-1wi22pg-Code"
  >
    code
  </code>
</div>
`;
